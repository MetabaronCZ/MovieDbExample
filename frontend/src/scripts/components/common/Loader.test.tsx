import { describe, expect, it } from '@jest/globals';

import { Loader } from './Loader';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/Loader', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Loader />, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const ico = container.getElementsByTagName('i');
    expect(ico.length).toEqual(1);
  });
});
