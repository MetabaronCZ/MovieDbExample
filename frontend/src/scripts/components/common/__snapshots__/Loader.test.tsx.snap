// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Loader should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 16px;
  align-items: center;
  justify-content: center;
}

.c1 {
  padding: 8px 0;
}

.c2 {
  display: inline-block;
  width: 24px;
  height: 24px;
  border-width: 2px;
  border-style: solid;
  border-color: #98c3d9;
  border-top-color: #002d56;
  border-radius: 50%;
  animation: kJPzGr 1s ease-in-out infinite;
}

<div>
  <div
    class="c0 c1"
  >
    <i
      class="c2"
    />
  </div>
</div>
`;
